tests/test_integrals.py

```diff
@@ -1479,3 +1512,102 @@ def fun2(pitch):
+        data, things = TestBounce1D.get_drift_analytic_data()
+        # Compute analytic approximation.
+        drift_analytic, _, _, pitch_inv = TestBounce1D.drift_analytic(data)
+
+        # Recompute on non-symmetric, fft compatible grid.
+        eq = things["eq"]
+        # FIXME: Change LinearGrid to default to Fourier points nodes. Interpolation
+        #        fails on LinearGrid.
+        grid = Grid.create_meshgrid(
+            [data["rho"], fourier_pts(4 * eq._M_grid), fourier_pts(1) / eq.NFP],
+            NFP=eq.NFP,
+        )
+        grid_data = eq.compute(
+            names=Bounce2D.required_names + ["cvdrift", "gbdrift"], grid=grid
+        )
+        grid_data["cvdrift"] = grid_data["cvdrift"] * data["normalization"]
+        grid_data["gbdrift"] = grid_data["gbdrift"] * data["normalization"]
+
+        # Compute numerical result.
+        M, N = 512, 16
+        # I have concern that numpy computes Nyquist frequency component incorrectly,
+        # and the reason high Fourier resolution is required is that with large sample
+        # frequency the incorrect max frequency component is higher than the max
+        # frequency of theta, hiding the mistake.
+        bounce = Bounce2D(
+            grid=grid,
+            data=grid_data,
+            theta=Bounce2D.compute_theta(
+                eq,
+                data["rho"],
+                M=M,
+                N=N,
+                iota=jnp.broadcast_to(data["iota"], shape=(M * N)),
+            ),
+            N_B=2 * N,
+            num_transit=4,
+            alpha=data["alpha"],  # - 2 * np.pi * data["iota"],
+            Bref=data["Bref"],
+            Lref=data["a"],
+            check=True,
+            plot=True,
+        )
+        points = bounce.points(pitch_inv, num_well=1)
+        bounce.check_points(points, pitch_inv, plot=True)
+
+        f = Bounce2D.reshape_data(grid, grid_data["cvdrift"], grid_data["gbdrift"])
+        drift_numerical_num = bounce.integrate(
+            integrand=TestBounce1D.drift_num_integrand,
+            points=points,
+            pitch_inv=pitch_inv,
+            f=f,
+            check=True,
+            plot=True,
+        )
+        drift_numerical_den = bounce.integrate(
+            integrand=TestBounce1D.drift_den_integrand,
+            points=points,
+            pitch_inv=pitch_inv,
+            check=True,
+            plot=True,
+        )
+        drift_numerical = np.squeeze(drift_numerical_num / drift_numerical_den)
+        msg = "There should be one bounce integral per pitch in this example."
+        assert drift_numerical.size == drift_analytic.size, msg
+        #
+        # np.testing.assert_allclose(  # noqa: E800
+        #     drift_numerical, drift_analytic, atol=5e-3, rtol=5e-2  # noqa: E800
+        # )  # noqa: E800
+
+        fig, ax = plt.subplots()
+        ax.plot(pitch_inv, drift_analytic)
+        ax.plot(pitch_inv, drift_numerical)
+        plt.show()
```

Comment:
failing for some reason, but it's difficult to isolate the cause.

For debugging, I've been 
1. Changing `zeta = np.linspace(-np.pi / iota, np.pi / iota, (2 * eq.M_grid) * 4 + 1)` in `get_drift_analytic_data` to shift to `zeta = np.linspace(np.pi / iota, 3 * np.pi / iota, (2 * eq.M_grid) * 4 + 1)` just to ensure an apples to apples comparison. Also commenting out the assertions in  `get_drift_analytic_data` 
2. Setting `plot=True` in the ``test_binormal_drift_bounce1d``
```python
# line 1410
drift_numerical_num = bounce.integrate(
    integrand=TestBounce1D.drift_num_integrand,
    points=points,
    pitch_inv=pitch_inv,
    f=f,
    check=True,
    plot=True,
)
```
3. Comparing plots. I notice that all the bounce points are computed the same, the plots of |B| and all quantities that are interpolated to the integrand right before the quadrature are identical. Then the quadrature is performed and we see from the final plots that `Bounce2D` differs from `Bounce1D` and the analytic approximation by some factor that increases linearly with the pitch angle.